plutus-ledger-api/src/PlutusLedgerApi/Common/SerialisedScript.hs

```diff
@@ -133,48 +146,33 @@ scriptCBORDecoder lv pv =
         (p :: UPLC.Program UPLC.FakeNamedDeBruijn DefaultUni DefaultFun ()) <- decodeViaFlat flatDecoder
         pure $ coerce p
 
--- | The deserialization from a serialised script to a Script (for execution).
--- Called inside phase-1 validation (assertScriptWellFormed) and inside phase-2's `mkTermToEvaluate`
-fromSerialisedScript :: forall e m. (AsScriptDecodeError e, MonadError e m)
+-- | The deserialization from a serialised script into a `ScriptForEvaluation`,
+-- ready to be evaluated on-chain.
+-- Called inside phase-1 validation (i.e., deserialisation error is a phase-1 error).
+deserialiseScript :: forall m. (MonadError ScriptDecodeError m)
                      => PlutusLedgerLanguage -- ^ the Plutus ledger language of the script.
                      -> ProtocolVersion -- ^ which protocol version the script was submitted in.
```

Comment:
I think we could just change our `ProtocolVersion` type to be `MajorProtocolVersion`. It sounds like the basic situation is that `plutus` should never care about the minor protocol version, so we can represent that in the types.

Reply:
> I think we could just change our ProtocolVersion type to be MajorProtocolVersion.

I'll open a separate PR for this.

> should not change for intra-era hardforks

Do you mean "should not be more restrictive"? It can become more permissive as new builtins or new Plutus Core versions are added in an intra-era HF.

> since ledger will not be able to supply to this function a major protocol version that was bumped during an intra-era hardfork.

Can you elaborate? Are you saying we'll still get the old major PV after an intra-era HF? Why is that?